be/src/storage/rowset/beta_rowset_writer.cpp

```diff
@@ -488,7 +497,12 @@ Status HorizontalBetaRowsetWriter::_final_merge() {
 Status HorizontalBetaRowsetWriter::_flush_segment_writer(std::unique_ptr<SegmentWriter>* segment_writer) {
     uint64_t segment_size;
     uint64_t index_size;
-    RETURN_IF_ERROR((*segment_writer)->finalize(&segment_size, &index_size));
+    uint64_t footer_position;
+    RETURN_IF_ERROR((*segment_writer)->finalize(&segment_size, &index_size, &footer_position));
+    uint64_t footer_size = segment_size - footer_position;
+    auto* partial_rowset_footer = _rowset_txn_meta_pb.add_partial_rowset_footers();
```

Comment:
make _rowset_txn_meta_pb a unique_ptr, and only set fields when it's a partial update